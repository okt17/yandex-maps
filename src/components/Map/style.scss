.app__map {
  height: 100%;
}
