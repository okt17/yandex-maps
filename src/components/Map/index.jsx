import React from 'react';
import PropTypes from 'prop-types';
import './style';

class Map extends React.Component {
  static propTypes = {
    initMap: PropTypes.func.isRequired,
  };
  ref = React.createRef();
  componentDidMount() {
    this.props.initMap(this.ref.current);
  }
  shouldComponentUpdate() {
    return false;
  }
  render() {
    return <div className='app__map' ref={this.ref} />;
  }
}

export default Map;
