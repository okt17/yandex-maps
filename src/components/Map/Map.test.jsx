/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Map from '.';

test('Map component test', () => {
  const component = renderer.create(<Map
    initMap={() => {}}
  />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
