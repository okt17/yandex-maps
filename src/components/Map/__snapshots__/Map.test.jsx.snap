// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component test 1`] = `
<div
  className="app__map"
/>
`;
